portal-api/src/v1/controllers/utilisation-report-service/previous-reports.controller.js

```diff
@@ -0,0 +1,66 @@
+const api = require('../../api');
+
+const getMonthName = (monthNumber) => {
+  // date is set to 1st Jan to avoid bug when today's month has more days than target month
+  const date = new Date(2023, 1, 1);
+  // offset by 1 as January = 1 in Database
+  date.setMonth(monthNumber - 1);
+  return date.toLocaleString('default', { month: 'long' });
+};
+
+const getYears = (reports) => {
+  const years = reports.map((report) => report.year);
+  return [...new Set(years)];
+};
+
+const getReportsGroupedByYear = (years, reports) => years.map((year) => {
+  const reportsByYear = reports
+    .filter((report) => report.year === year)
+    .map((report) => ({
+      month: getMonthName(report.month),
+      path: report.path
+    }));
+  return {
+    year,
+    reports: reportsByYear,
+  };
+});
+
+const populateOmittedYears = (reportsGroupedByYear, years) => {
```

Comment:
Minor nit-pick, can this return the array back so we reassign to a new variable in the function below please? little more obvious than a side effect of changing the array passed in etc.

Reply:
On it